aptos-move/aptos-aggregator/src/aggregator_change_set.rs

```diff
@@ -0,0 +1,507 @@
+            max_value: 100,
+            state: AggregatorState::Delta {
+                speculative_start_value: SpeculativeStartValue::AggregatedValue(20),
+                delta: DeltaValue::Positive(10),
+                history: DeltaHistory {
+                    max_achieved_positive_delta: 30,
+                    min_achieved_negative_delta: 15,
+                    min_overflow_positive_delta: Some(90),
+                    max_underflow_negative_delta: Some(25),
+                },
+            },
+            base_aggregator: None,
+        };
+        let mut aggregator_change2 = AggregatorChange {
+            max_value: 100,
+            state: AggregatorState::Delta {
+                speculative_start_value: SpeculativeStartValue::AggregatedValue(40),
+                delta: DeltaValue::Positive(20),
+                history: DeltaHistory {
+                    max_achieved_positive_delta: 25,
+                    min_achieved_negative_delta: 20,
+                    min_overflow_positive_delta: Some(95),
+                    max_underflow_negative_delta: Some(45),
+                },
+            },
+            base_aggregator: None,
+        };
+        assert_ok!(aggregator_change2.merge_with_previous_aggregator_change(aggregator_change1));
+        assert_eq!(aggregator_change2, AggregatorChange {
+            max_value: 100,
+            state: AggregatorState::Delta {
+                speculative_start_value: SpeculativeStartValue::Unset,
+                delta: DeltaValue::Positive(30),
+                history: DeltaHistory {
+                    max_achieved_positive_delta: 35,
+                    min_achieved_negative_delta: 15,
+                    min_overflow_positive_delta: Some(90),
+                    max_underflow_negative_delta: Some(25),
+                },
+            },
+            base_aggregator: None,
+        });
+    }
+
+    #[test]
+    fn test_merge_delta_into_delta2() {
+        let aggregator_change1 = AggregatorChange {
+            max_value: 100,
+            state: AggregatorState::Delta {
+                speculative_start_value: SpeculativeStartValue::AggregatedValue(70),
+                delta: DeltaValue::Negative(40),
+                history: DeltaHistory {
+                    max_achieved_positive_delta: 20,
+                    min_achieved_negative_delta: 60,
+                    min_overflow_positive_delta: Some(40),
+                    max_underflow_negative_delta: Some(80),
+                },
+            },
+            base_aggregator: None,
+        };
+        let mut aggregator_change2 = AggregatorChange {
+            max_value: 100,
+            state: AggregatorState::Delta {
+                speculative_start_value: SpeculativeStartValue::AggregatedValue(60),
+                delta: DeltaValue::Negative(20),
+                history: DeltaHistory {
+                    max_achieved_positive_delta: 35,
+                    min_achieved_negative_delta: 20,
```

Comment:
isn't it better to test when 20+40 != 60, i.e. wouldn't it be choosing between 60 and 60 here.